python/ray/data/tests/test_size_estimation.py

```diff
@@ -113,6 +116,135 @@ def test_arrow_size_add_block(ray_start_regular_shared):
     assert b2.build().num_rows == 10000
 
 
+def test_split_read_csv(ray_start_regular_shared, tmp_path):
+    ctx = ray.data.context.DatasetContext.get_current()
+
+    def gen(name):
+        path = os.path.join(tmp_path, name)
+        ray.data.range(1000, parallelism=1).map(lambda _: LARGE_VALUE) \
+            .write_csv(path)
+        return ray.data.read_csv(path)
+
+    # 20MiB
+    ctx.target_max_block_size = 20_000_000
+    ds1 = gen("out1")
+    assert ds1._block_num_rows() == [1000]
+
+    # 3MiB
+    ctx.target_max_block_size = 3_000_000
+    ds2 = gen("out2")
+    nrow = ds2._block_num_rows()
+    assert 3 < len(nrow) < 5, nrow
```

Comment:
`len(nrow) == 4`?

Reply:
It's just in case the size differs in minor ways in different platforms.